internal/audit/cerboshub/cerboshub.go

```diff
@@ -171,27 +181,52 @@ func schedule(db *badgerv4.DB, muTimer *mutexTimer, syncer IngestSyncer, minFlus
 	// This prevents a retry completion occurring immediately before the next sync
 	// (and therefore burdening the backend).
 	muTimer.set(minFlushInterval)
+
+	select {
+	case errCh <- err:
```

Comment:
Won't this `err` be `nil` in most cases?

Reply:
Yeah, I was on the edge of using a `chan struct{}` but figured err might be more idiomatic/future-proof. Perhaps pointless

Reply:
It's used for both testing and waiting for `sync` completion on shutdown. The [`ForceWrite`](https://github.com/cerbos/cerbos/blob/f63a63d2a0194d731b29ae5dfdff416a7b5e1868/internal/audit/local/badgerdb.go#L189-L198) method closes the buffer and [triggers a call to the callback](https://github.com/cerbos/cerbos/blob/f63a63d2a0194d731b29ae5dfdff416a7b5e1868/internal/audit/local/badgerdb.go#L131-L137). The error channel just means we can block until we're sure the logs have been synced. Similarly [on shutdown](https://github.com/cerbos/cerbos/blob/f63a63d2a0194d731b29ae5dfdff416a7b5e1868/internal/audit/local/badgerdb.go#L126-L130). The test case in particular means that we can do away with any arbitrary and non-guaranteed `Sleep`s in the test code.

In it's existing form, yeah -- it could totally just be a `chan struct{}`.

> which means that there's a small chance that the `awaitCallbackFn` hangs forever because it's reading from an empty channel

I figured this was safe "enough" given the close locality. E.g. you can only wait for channel closure if you create the channel to pass to the callback?

Reply:
I see. Makes sense. Since the channel has a size of 1, the `select` for writing to it is probably not necessary? Also might be better to change the type of the callback to `func(chan <-struct{})` to make sure that it's write-only for the callback.

Reply:
OK this is probably _really_ silly, but the reason for the `select` was just in case the channel creator initialised an unbuffered channel. At this point I'm not sure which component should be inferring these requirements 🤪

Reply:
I totally contradicted myself here:

> I figured this was safe "enough" given the close locality. E.g. you can only wait for channel closure if you create the channel to pass to the callback?

> just in case the channel creator initialised an unbuffered channel

Unbuffered channel with nicely isolated code block seems reasonable enough:

```go
ch := make(chan struct{})
go l.callbackFn(ch)
<-ch
```